src/runtime_src/core/tools/xbmgmt2/SubCmdProgram.cpp

```diff
@@ -58,11 +57,41 @@ namespace po = boost::program_options;
 
 namespace {
 /*
- * Update shell on the board
+ * Update shell on the board for auto flash
+ */
+static void 
+update_shell(uint16_t index, const std::string& primary, const std::string& secondary)
+{
+  std::unique_ptr<firmwareImage> pri;
+  std::unique_ptr<firmwareImage> sec;
```

Comment:
Better to move declarations with definitions or closer to where they are used.
In this case line 65 can be removed and line 75 can be prefixed with auto.
Line 66 can move below line 75.
This makes it clearer where the variables are used.
Same comments for other parts of code.